lib/spack/spack/spec.py

```diff
@@ -2257,8 +2478,14 @@ def _concretize_helper(self, concretizer, presets=None, visited=None):
         changed = False
 
         # Concretize deps first -- this is a bottom-up process.
-        for name in sorted(self._dependencies.keys()):
-            changed |= self._dependencies[name].spec._concretize_helper(
+        for name in sorted(self._dependencies):
+            # WARNING: This function is an implementation detail of the
+            # WARNING: original concretizer. Since with that greedy
+            # WARNING: algorithm we don't allow multiple nodes from
+            # WARNING: the same package in a DAG, here we hard-code
+            # WARNING: using index 0 i.e. we assume that we have only
+            # WARNING: one edge from package "name"
```

Comment:
When will we ever (even now) have two *dependencies* with the same name?

Reply:
resolving per discussion points below.

Reply:
Closing this, see https://github.com/spack/spack/pull/28673#issuecomment-1048557787